libraries/AP_Camera/AP_Camera_Backend.h

```diff
@@ -138,6 +145,12 @@ class AP_Camera_Backend
         int16_t num_remaining;      // number of pictures still to be taken, -1 means take unlimited pictures
     } time_interval_settings;
 
+    struct {
+        HAL_Semaphore sem;        // semaphore
+        Location loc;             // location of the vehicle on which poi is calculated
```

Comment:
let's add these fields:

- "uint8_t request_id" - this will be incremented by the main code when send_camera_fov_status() is called
- "uint32_t poi_loc_update_ms" - this will be incremented by the thread whenever it has successfully update the poi_loc
- "Location camera_loc" - this will be set by the thread when it first starts a new calculation
- "Quaternion camera_attitude" - this will be set by the thread (using the appropriate mount orientation) when it first starts a new calculation
- BTW the camera fov fields can be filled in by the send_camera_fov_status method.